app/controllers/concerns/dfe_authentication.rb

```diff
@@ -0,0 +1,45 @@
+module DFEAuthentication
+  extend ActiveSupport::Concern
+
+  included do
+    before_action :set_current_user
+
+    def current_user
+      @current_user ||= session[:current_user]
+    end
+    alias_method :set_current_user, :current_user
+
+    def user_signed_in?
+      current_user.present?
+    end
+
+    def require_auth
+      unless user_signed_in?
```

Comment:
Really subjective but I'd be inclined to `return unless user_signed_in?` instead of indenting the rest of the block?